### PR TITLE
path-of-building: add desktop item

### PR DESCRIPTION
**Pull request by iynaix**
## Description of changes

Added a desktop item for pob so `pob://` urls on poe ninja etc can be opened directly.
Also formatted with nixfmt-rfc-style.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/path-of-building/default.nix [hunk 1]
```diff
@@ -59,6 +85,31 @@ stdenv.mkDerivation {
...
+    (makeDesktopItem {
+      name = "Path of Building";
+      desktopName = "Path of Building";
+      comment = "Offline build planner for Path of Exile";
+      exec = "pobfrontend %U";
+      terminal = false;
+      type = "Application";
+      icon = fetchurl {
+        name = "PathOfBuilding-logo.png";
+        url = "https://avatars.githubusercontent.com/u/60926656?s=200&v=4";
```
Comment (K900) [inline pkgs/games/path-of-building/default.nix:98]: That's definitely not going to be stable long term.

Comment (jopejoe1) [reply pkgs/games/path-of-building/default.nix:98]: Also, icon should be an icon name ad not a path, so that icon themes work on it.

Comment (iynaix) [reply pkgs/games/path-of-building/default.nix:98]: I was unable to find an icon within the pob repo (I think it's embedded within the .exe), the closest I managed to find was the favicon within the repo for the website: https://raw.githubusercontent.com/PathOfBuildingCommunity/pathofbuildingcommunity.github.io/master/favicon.ico

Comment (jopejoe1) [reply pkgs/games/path-of-building/default.nix:98]: You can use imagemagick's convert to convert it to png.

Comment (iynaix) [reply pkgs/games/path-of-building/default.nix]: Ok, I've modified it to extract the icons from the exe file directly, no longer need to download the icon from somewhere else. :)

File: pkgs/games/path-of-building/default.nix [hunk 2]
```diff
@@ -59,6 +97,27 @@ stdenv.mkDerivation {
     )
   '';
 
+  desktopItems = [
+    (makeDesktopItem {
+      name = "Path of Building";
```
Comment (K900) [inline pkgs/games/path-of-building/default.nix:102]: The name should probably be `path-of-building` or `pobfrontend` or something without spaces.

Comment (iynaix) [reply pkgs/games/path-of-building/default.nix]: Ok, done.

**Comment by K900** [comment]
Seems good, thanks!

**Pull request merged**